File: Tutorial2-1InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/di/MyComponent.java
package com.example.tutorial2injectconstructor.di;

import com.example.tutorial2injectconstructor.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = SharedPreferencesModule.class)
@Singleton
public interface MyComponent {
    void inject(MainActivity mainActivity);
}

File: Tutorial2-1InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/di/MyComponent.kt
package com.example.tutorial2injectconstructor.di

import com.example.tutorial2injectconstructor.MainActivity
import dagger.Component
import javax.inject.Singleton

@Component(modules = [SharedPreferencesModule::class])
@Singleton
interface MyComponent {
    fun inject(mainActivity: MainActivity)
}